src/pyhf/infer/test_statistics.py

```diff
@@ -58,5 +58,5 @@ def qmu(mu, data, pdf, init_pars, par_bounds):
     qmu = fixed_poi_fit_lhood_val - unconstrained_fit_lhood_val
     qmu = tensorlib.where(
         muhatbhat[pdf.config.poi_index] > mu, tensorlib.astensor(0.0), qmu
-    )[0]
+    )
```

Comment:
This change is so that JAX doesn't have a problem, as trying to access the `[0]` element for a 0-d JAX DeviceArray will cause the error
```
IndexError: Too many indices for array: 1 non-None/Ellipsis indices for dim 0.
```

This is unfortunate, as `qmu` get called less often then the optimizers.

Reply:
Why is htis unfortunate if it gets covered by tests?

Reply:
> Why is htis unfortunate if it gets covered by tests?

Shouldn't this be _barely_ slower now as we're having to perform slices more often?

Reply:
Not necessarily. Slices are views.